infullmvp-kotlin-basetest/src/main/java/com/infullmobile/android/infullmvp/basetest/InFullMvpCardBaseTest.kt

```diff
@@ -0,0 +1,47 @@
+package com.infullmobile.android.infullmvp.basetest
+
+import com.infullmobile.android.infullmvp.InFullMvpActivity
+import com.infullmobile.android.infullmvp.InFullMvpView
+import com.infullmobile.android.infullmvp.PresentedCustomView
+import com.infullmobile.android.infullmvp.Presenter
+import org.junit.Before
+import org.robolectric.Robolectric
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.android.controller.ActivityController
+
+abstract class InFullMvpCardBaseTest<
+        T : InFullMvpView<PresenterType, PresentedViewType>,
+        PresenterType : Presenter<PresentedViewType>,
+        out PresentedViewType : PresentedCustomView<PresenterType>,
+        ActivityType : InFullMvpActivity<*, *>> {
```

Comment:
Not sure why we need the activty type. The containing activity shouldn't even need to be of InFullMvpActivity type if you ask me.

Reply:
If we'll build normal Activity we'll get this error in SampleMvpCardGraph
android.app.Activity cannot be cast to com.infullmobile.android.infullmvp.sample.di_components.HasComponent